app/backend/approaches/chatreadretrieveread.py

```diff
@@ -127,61 +129,60 @@ async def run_until_final_call(
         minimum_search_score = overrides.get("minimum_search_score", 0.0)
         minimum_reranker_score = overrides.get("minimum_reranker_score", 0.0)
         filter = self.build_filter(overrides, auth_claims)
+        current_model = self.hf_model if self.hf_model else self.chatgpt_model
 
         original_user_query = messages[-1]["content"]
         if not isinstance(original_user_query, str):
             raise ValueError("The most recent message content must be a string.")
-        user_query_request = "Generate search query for: " + original_user_query
 
-        tools: List[ChatCompletionToolParam] = [
-            {
-                "type": "function",
-                "function": {
-                    "name": "search_sources",
-                    "description": "Retrieve sources from the Azure AI Search index",
-                    "parameters": {
-                        "type": "object",
-                        "properties": {
-                            "search_query": {
-                                "type": "string",
-                                "description": "Query string to retrieve documents from azure search eg: 'Health care plan'",
-                            }
-                        },
-                        "required": ["search_query"],
-                    },
-                },
-            }
-        ]
+        # Process results
```

Comment:
I see that we've moved these tools inside the templates. What are they? What are they used for?

Reply:
From my understanding, tools are not that important as the larger LLMs can do really well with text, though the tool should 'kind of' help with providing consistent results, in this case extracting and creating a query that would be suitable for querying Azure Search.